test/vtdb_test.py

```diff
@@ -375,6 +398,38 @@ def test_tablet_restart_stream_execute(self):
       self.fail("Communication with shard0 replica failed with error %s" %
                 str(e))
 
+  def test_fail_mid_stream_execute(self):
+    """Tests for errors in the middle of a stream execute.
+
+    This is for testing app failures, not infrastructure failures. We simulate
+    an app failure by killing MySQL on the master tablet.
+
+    This test is inherently racy, as it tries to kill MySQL in the middle
```

Comment:
I meant StreamExecute("select sleep(30)") or whatever the syntax is for 30 seconds. You'll get the Fields, then nothing for 30 seconds. If you sleep for 10s and then kill mysqld, should not be racy at all.